src/test/java/APITests/RestAssuredDemoTest.java

```diff
@@ -246,8 +246,30 @@ public void testListUserPublicPosts() {
         int statusCode = response.statusCode();
         Assert.assertEquals(statusCode, HttpStatus.SC_OK);
         Assert.assertEquals(response.jsonPath().get("postStatus").toString(), "[public]");
-
     }
 
+    @Test(dependsOnGroups = "signup", dependsOnMethods = {"testListUserPublicPosts", "testCreatePublicPost"})
+    public void testDeleteUserPost() {
+        String postId = "postId";
```

Comment:
Here you create local variable and assign the value "postId" - to make it working make the variable global like the userId and assign the value of the post you have created for it.